hydra-cardano-api/src/Hydra/Cardano/Api/TxBody.hs

```diff
@@ -63,6 +65,32 @@ findScriptMinting (getTxBody -> ShelleyTxBody _ body scripts _ _ _) policyId = d
   needle = toLedgerScriptHash policyId
   haystack = getField @"minted" body :: Set (Ledger.ScriptHash StandardCrypto)
 
+foldRedeemers ::
+  Tx AlonzoEra ->
+  result ->
+  (RdmrTag -> Word64 -> ScriptData -> ExecutionUnits -> result -> result) ->
+  result
+foldRedeemers (getTxBody -> ShelleyTxBody _ _ _ scriptData _ _) result0 step =
+  Map.foldrWithKey
+    ( \(Ledger.RdmrPtr tag ix) (d, ex) ->
+        step (fromLedgerTag tag) ix (fromLedgerData d) (fromLedgerExUnits ex)
+    )
+    result0
+    redeemers
+ where
+  redeemers = case scriptData of
+    TxBodyNoScriptData ->
+      mempty
+    TxBodyScriptData _ _ (Ledger.Redeemers rs) ->
+      rs
+
+hasScript ::
+  Tx AlonzoEra ->
+  PlutusScript lang ->
+  Bool
+hasScript (getTxBody -> ShelleyTxBody _ _ scripts _ _ _) script =
+  script `elem` (fromLedgerScript <$> scripts)
```

Comment:
According to your "where to locate code policy" these two functions do not belong here but in a `Tx` module?

Reply:
Yes, actually, this whole module has misplaced functions. I intended to get rid of that later.